[Bug]: `get_data_in_units` fails for ElectricalSeries with `channel_conversion`

### What happened?

https://github.com/NeurodataWithoutBorders/pynwb/pull/1745 added a method to get the data in the corect units which is something very nice that I always wanted. 

One corner case that i think is not included is that the method will produce incorrect results with an `ElectricalSeries` object that has a channel conversion.

### Steps to Reproduce

This fails on the dev branch:


```python
from pynwb.testing.mock.ecephys import mock_ElectricalSeries

import numpy as np

data = np.asarray([[1, 1, 1, 1, 1], [1, 1, 1, 1, 1]])
conversion = 1.0
offset = 3.0
channel_conversion = np.asarray([2.0, 2.0])
electrical_series = mock_ElectricalSeries(
    data=data,
    conversion=conversion,
    offset=offset,
    channel_conversion=channel_conversion,
)

data_in_units = electrical_series.get_data_in_units()
expected_data =  data * conversion * channel_conversion[:, np.newaxis] + offset

np.testing.assert_almost_equal(data_in_units, expected_data)
```


### Traceback

```
AssertionError: 
Arrays are not almost equal to 7 decimals

Mismatched elements: 10 [/](https://file+.vscode-resource.vscode-cdn.net/) 10 (100%)
Max absolute difference: 1.
Max relative difference: 0.2
 x: array([[4., 4., 4., 4., 4.],
       [4., 4., 4., 4., 4.]])
 y: array([[5., 5., 5., 5., 5.],
       [5., 5., 5., 5., 5.]])
```

### Operating System

Windows

### Python Executable

Conda

### Python Version

lower version (unsupported)

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,7 +8,7 @@
 - Add `pynwb.get_nwbfile_version()`. @bendichter [#1703](https://github.com/NeurodataWithoutBorders/pynwb/pull/1703)
 - Updated timeseries data checks to warn instead of error when reading invalid files. @stephprince [#1793](https://github.com/NeurodataWithoutBorders/pynwb/pull/1793)
 - Expose the offset, conversion and channel conversion parameters in `mock_ElectricalSeries`. @h-mayorquin [#1796](https://github.com/NeurodataWithoutBorders/pynwb/pull/1796)
-- Enchance `get_data_in_units()` to work with objects that have a `channel_conversion` attribute like the `ElectricalSeries`  @h-mayorquin [#1806](https://github.com/NeurodataWithoutBorders/pynwb/pull/1806)
+- Enhance `get_data_in_units()` to work with objects that have a `channel_conversion` attribute like the `ElectricalSeries`  @h-mayorquin [#1806](https://github.com/NeurodataWithoutBorders/pynwb/pull/1806)
 
 ### Bug fixes
 - Fix bug where namespaces were loaded in "w-" mode. @h-mayorquin [#1795](https://github.com/NeurodataWithoutBorders/pynwb/pull/1795)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@
 - Add `pynwb.get_nwbfile_version()`. @bendichter [#1703](https://github.com/NeurodataWithoutBorders/pynwb/pull/1703)
 - Updated timeseries data checks to warn instead of error when reading invalid files. @stephprince [#1793](https://github.com/NeurodataWithoutBorders/pynwb/pull/1793)
 - Expose the offset, conversion and channel conversion parameters in `mock_ElectricalSeries`. @h-mayorquin [#1796](https://github.com/NeurodataWithoutBorders/pynwb/pull/1796)
+- Enchance `get_data_in_units()` to work with objects that have a `channel_conversion` attribute like the `ElectricalSeries`  @h-mayorquin [#1806](https://github.com/NeurodataWithoutBorders/pynwb/pull/1806)
 
 ### Bug fixes
 - Fix bug where namespaces were loaded in "w-" mode. @h-mayorquin [#1795](https://github.com/NeurodataWithoutBorders/pynwb/pull/1795)

--- a/src/pynwb/base.py
+++ b/src/pynwb/base.py
@@ -296,7 +296,7 @@ class TimeSeries(NWBDataInterface):
             return np.arange(len(self.data)) / self.rate + self.starting_time
 
     def get_data_in_units(self):
-        return np.asarray(self.data) * self.conversion + self.offset
+        return np.asarray(self.data) * self.conversion * self.channel_conversion[:, np.newaxis] + self.offset
 
 
 @register_class('Image', CORE_NAMESPACE)

--- a/src/pynwb/base.py
+++ b/src/pynwb/base.py
@@ -297,7 +297,7 @@ class TimeSeries(NWBDataInterface):
 
     def get_data_in_units(self):
         if "channel_conversion" in self.fields:
-            scale_factor =  self.conversion * self.channel_conversion[:, np.newaxis]
+            scale_factor = self.conversion * self.channel_conversion[:, np.newaxis]
         else:
             scale_factor = self.conversion
         return np.asarray(self.data) * scale_factor + self.offset

--- a/src/pynwb/base.py
+++ b/src/pynwb/base.py
@@ -296,7 +296,11 @@ class TimeSeries(NWBDataInterface):
             return np.arange(len(self.data)) / self.rate + self.starting_time
 
     def get_data_in_units(self):
-        return np.asarray(self.data) * self.conversion * self.channel_conversion[:, np.newaxis] + self.offset
+        if "channel_conversion" in self.fields:
+            scale_factor =  self.conversion * self.channel_conversion[:, np.newaxis]
+        else:
+            scale_factor = self.conversion
+        return np.asarray(self.data) * scale_factor + self.offset
 
 
 @register_class('Image', CORE_NAMESPACE)

--- a/tests/unit/test_ecephys.py
+++ b/tests/unit/test_ecephys.py
@@ -131,9 +131,10 @@ def test_get_data_in_units():
     )
 
     data_in_units = electrical_series.get_data_in_units()
-    expected_data =  data * conversion * channel_conversion[:, np.newaxis] + offset
+    expected_data = data * conversion * channel_conversion[:, np.newaxis] + offset
 
     np.testing.assert_almost_equal(data_in_units, expected_data)
+
 
 class SpikeEventSeriesConstructor(TestCase):
 

--- a/tests/unit/test_ecephys.py
+++ b/tests/unit/test_ecephys.py
@@ -117,23 +117,23 @@ class ElectricalSeriesConstructor(TestCase):
                    ) in str(w[-1].message)
 
 
-def test_get_data_in_units():
+    def test_get_data_in_units(self):
 
-    data = np.asarray([[1, 1, 1, 1, 1], [1, 1, 1, 1, 1]])
-    conversion = 1.0
-    offset = 3.0
-    channel_conversion = np.asarray([2.0, 2.0])
-    electrical_series = mock_ElectricalSeries(
-        data=data,
-        conversion=conversion,
-        offset=offset,
-        channel_conversion=channel_conversion,
-    )
+        data = np.asarray([[1, 1, 1, 1, 1], [1, 1, 1, 1, 1]])
+        conversion = 1.0
+        offset = 3.0
+        channel_conversion = np.asarray([2.0, 2.0])
+        electrical_series = mock_ElectricalSeries(
+            data=data,
+            conversion=conversion,
+            offset=offset,
+            channel_conversion=channel_conversion,
+        )
 
-    data_in_units = electrical_series.get_data_in_units()
-    expected_data = data * conversion * channel_conversion[:, np.newaxis] + offset
+        data_in_units = electrical_series.get_data_in_units()
+        expected_data = data * conversion * channel_conversion[:, np.newaxis] + offset
 
-    np.testing.assert_almost_equal(data_in_units, expected_data)
+        np.testing.assert_almost_equal(data_in_units, expected_data)
 
 
 class SpikeEventSeriesConstructor(TestCase):

--- a/tests/unit/test_ecephys.py
+++ b/tests/unit/test_ecephys.py
@@ -18,6 +18,7 @@ from pynwb.ecephys import (
 from pynwb.device import Device
 from pynwb.file import ElectrodeTable
 from pynwb.testing import TestCase
+from pynwb.testing.mock.ecephys import mock_ElectricalSeries
 
 from hdmf.common import DynamicTableRegion
 
@@ -115,6 +116,24 @@ class ElectricalSeriesConstructor(TestCase):
                "but instead the first does. Data is oriented incorrectly and should be transposed."
                    ) in str(w[-1].message)
 
+
+def test_get_data_in_units():
+
+    data = np.asarray([[1, 1, 1, 1, 1], [1, 1, 1, 1, 1]])
+    conversion = 1.0
+    offset = 3.0
+    channel_conversion = np.asarray([2.0, 2.0])
+    electrical_series = mock_ElectricalSeries(
+        data=data,
+        conversion=conversion,
+        offset=offset,
+        channel_conversion=channel_conversion,
+    )
+
+    data_in_units = electrical_series.get_data_in_units()
+    expected_data =  data * conversion * channel_conversion[:, np.newaxis] + offset
+
+    np.testing.assert_almost_equal(data_in_units, expected_data)
 
 class SpikeEventSeriesConstructor(TestCase):
 

--- a/tests/unit/test_ecephys.py
+++ b/tests/unit/test_ecephys.py
@@ -116,7 +116,6 @@ class ElectricalSeriesConstructor(TestCase):
                "but instead the first does. Data is oriented incorrectly and should be transposed."
                    ) in str(w[-1].message)
 
-
     def test_get_data_in_units(self):
 
         data = np.asarray([[1, 1, 1, 1, 1], [1, 1, 1, 1, 1]])

